FWCore/Common/interface/OutputProcessBlockHelper.h

```diff
@@ -0,0 +1,76 @@
+#ifndef FWCore_Common_OutputProcessBlockHelper_h
+#define FWCore_Common_OutputProcessBlockHelper_h
+
+/** \class edm::OutputProcessBlockHelper
+
+\author W. David Dagenhart, created 28 December, 2020
+
+*/
+
+#include "DataFormats/Provenance/interface/ProvenanceFwd.h"
+
+#include <set>
+#include <string>
+#include <vector>
+
+namespace edm {
+
+  class ProcessBlockHelperBase;
+
+  class OutputProcessBlockHelper {
+  public:
+    std::vector<std::string> const& processesWithProcessBlockProducts() const {
+      return processesWithProcessBlockProducts_;
+    }
+
+    void updateAfterProductSelection(std::set<std::string> const& processesWithKeptProcessBlockProducts,
+                                     ProcessBlockHelperBase const&);
+
+    void fillCacheIndices(StoredProcessBlockHelper&) const;
+
+    bool productsFromInputKept() const { return productsFromInputKept_; }
+
+    static constexpr unsigned int invalidCacheIndex() { return 0xffffffff; }
+
+    ProcessBlockHelperBase const* processBlockHelper() const { return processBlockHelper_; }
+
+    // These are intended to be used only for testing purposes
+    std::vector<unsigned int> const& translateFromStoredIndex() const { return translateFromStoredIndex_; }
```

Comment:
Using a `friend` could allow you to enforce this.

Reply:
Done. I think I like it. A little strange to declare a class defined in FWCore/Integration as a friend in FWCore/Common, but I suppose there is not really any real dependence...